src/librustc_codegen_llvm/back/lto.rs

```diff
@@ -525,10 +566,34 @@ fn thin_lto(cgcx: &CodegenContext<LlvmCodegenBackend>,
             }));
         }
 
+        // Save the accumulated ThinLTO import information
+        if let Some(path) = import_map_path {
+            if let Err(err) = curr_import_map.save_to_file(&path) {
+                let msg = format!("Error while writing ThinLTO import data: {}", err);
+                return Err(write::llvm_err(&diag_handler, &msg));
+            }
+        }
+
         Ok((opt_jobs, copy_jobs))
     }
 }
 
+/// Given two slices, each with no repeat elements. returns true if and only if
+/// the two slices have the same contents when considered as sets (i.e. when
+/// element order is disregarded).
+fn equivalent_as_sets(a: &[String], b: &[String]) -> bool {
+    // cheap path: unequal lengths means cannot possibly be set equivalent.
+    if a.len() != b.len() { return false; }
+    // fast path: before sorting, check if inputs are equivalent as is.
+    if a == b { return true; }
```

Comment:
You could also create two `FxHashSet`s and compare those.

Reply:
i debated about that. I guess it would make sense in terms of expected asymptotic behavior, right? (And its just a question of how the constant factors associated with hash-set construction compare to sorting a vector of str)

Reply:
I'll go ahead and put FxHashSet in. I'm not in the game of trying to microoptimize this beyond what I've already done.